src/zarr/core/sync.py

```diff
@@ -89,6 +90,20 @@ def cleanup_resources() -> None:
 atexit.register(cleanup_resources)
 
 
+def reset_resources_after_fork() -> None:
+    """
+    Ensure that global resources are reset after a fork. Without this function,
+    forked processes will retain invalid references to the parent process's resources.
+    """
+    global loop, iothread, _executor
+    loop[0] = None
+    iothread[0] = None
+    _executor = None
```

Comment:
Fix coverage: the main process never sees this function getting called. I suppose a test could explicitly call it.

Reply:
I will just exempt these lines from coverage, since they are being tested indirectly